virttest/env_process.py

```diff
@@ -1255,6 +1255,25 @@ def preprocess(test, params, env):
         version_info["vm_bootloader_version"] = str(vm_bootloader_ver)
         LOG.debug("vm bootloader version: %s", vm_bootloader_ver)
 
+    # Checking required virtio-win version, if not satisfied, cancel test
+    if params.get("required_virtio_win") or params.get("required_virtio_win_prewhql"):
+        if params.get("cdrom_virtio"):
+            cdrom_virtio = params["cdrom_virtio"]
+            virtio_win_range = params.get("required_virtio_win_prewhql") if re.search(
+                'prewhql', cdrom_virtio) else params.get("required_virtio_win")
+            if virtio_win_range:
+                LOG.info("Checking required virtio-win version: %s" % virtio_win_range)
+                cdrom_virtio_path = os.path.basename(utils_misc.get_path(data_dir.get_data_dir(), cdrom_virtio))
+                match = re.search("virtio-win-(?:prewhql-)?(\d+\.\d+(?:\.\d+)?-\d+)", cdrom_virtio_path)
+                if match.group(1) is None:
+                    test.error("Can not get virtio-win version from \"cdrom_virtio\": %s" % cdrom_virtio)
+                cdrom_virtio_version = re.sub('-', '.', match.group(1))
+                if cdrom_virtio_version not in VersionInterval(virtio_win_range):
+                    test.cancel("Got virtio-win version:%s, which is not in %s" %
+                                (cdrom_virtio_version, virtio_win_range))
```

Comment:
Hi @yanan-fu 

1.The current code logic is that if there is no available virtio_win_range for the current cdrom_virtio, this checkpoint will be skipped because the limitations of the virtio_win version are not suitable for the test's cdrom_virtio. 
 I would prefer to report a error if the configuration is wrong, because it is an error in the cfg, not a failure.What do you think?

2.If the cdrom_virtio not in the range, test will cancel.

Thanks a lot.

Reply:
> Hi @yanan-fu
> 
> 1.The current code logic is that if there is no available virtio_win_range for the current cdrom_virtio, this checkpoint will be skipped because the limitations of the virtio_win version are not suitable for the test's cdrom_virtio. I would prefer to report a error if the configuration is wrong, because it is an error in the cfg, not a failure.What do you think?

Yes, the new added L1274~L1275 is what i expect,  raise error when cdrom type and version type does not match,  instead of continue test.

test.error is suitable.        


> 
> 2.If the cdrom_virtio not in the range, test will cancel.
> 
> Thanks a lot.